packages/main/src/AvatarGroup.js

```diff
@@ -0,0 +1,456 @@
+	 * @private
+	 */
+	_prepareAvatars() {
+		const RTL = this.effectiveDir === "rtl";
+		this._colorIndex = 0;
+
+		this.items.forEach((avatar, index) => {
+			const colorIndex = this._colorBackgroundIndex;
+			avatar.interactive = !this._isGroup;
+
+			if (!avatar.getAttribute("background-color")) {
+				// AvatarGroup respects colors set to ui5-avatar
+				avatar.setAttribute("_background-color", AvatarBackgroundColor[`Accent${colorIndex}`]);
+			}
+
+			if (!avatar.getAttribute("size")) {
+				// AvatarGroup respects sizes set to ui5-avatar
+				avatar.setAttribute("_size", this.avatarSize);
+			}
+
+			// last avatar should not be offset as it breaks the container width and focus styles are no set correctly
+			if (index !== this._itemsCount - 1) {
+				// based on RTL margin left or right is set to avatars
+				avatar.style[`margin-${RTL ? "left" : "right"}`] = offsets[avatar.effectiveSize][this.type];
+			}
+		});
+	}
+
+	_onfocusin(event) {
+		const target = event.target;
+		this._itemNavigation.update(target);
+	}
+
+	/**
+	 * Returns the total width to item excluding the item width
+	 * RTL/LTR aware
+	 * @private
+	 * @param {HTMLElement} item
+	 */
+	_getWidthToItem(item) {
+		const isRTL = this.effectiveDir === "rtl";
+
+		if (isRTL) {
+			// in RTL the total width is equal to difference of the parent container width and
+			// how much is the item offset to the left minus its offsetWidth
+			return item.offsetParent.offsetWidth - item.offsetLeft - item.offsetWidth;
+		}
+
+		// in LTR the width is equal to item.offsetLeft
+		return item.offsetLeft;
+	}
+
+	/**
+	 * Overflows items that were not able to fit the container
+	 * @private
+	 */
+	_overflowItems() {
+		if (this.items.length < 2) {
+			// no need to overflow avatars
+			return;
+		}
+
+		const containerWidth = this._rootContainer.offsetWidth;
+		let hiddenItems = 0;
+
+		for (let index = 0; index < this._itemsCount; index++) {
+			const item = this.items[index];
+
+			// show item to determinate if it will fit the new container size
+			item.hidden = false;
+
+			// container width to current item + item width (avatar)
+			// used to determinate whether the following items will fit the container or not
+			let totalWidth = this._getWidthToItem(item) + item.offsetWidth;
+
+			if (index !== this._itemsCount - 1) {
+				totalWidth += this._overflowButtonEffectiveWidth;
+			}
+
+			if (totalWidth > containerWidth) {
+				hiddenItems = this._itemsCount - index;
+				break;
+			}
+		}
+
+		if (hiddenItems === 1) {
+			// show overflow button with at least 2 items
+			hiddenItems++;
+		}
+
+		// hide the items that did not fit the container size
+		this._setHiddenItems(hiddenItems);
+	}
+
+	_setHiddenItems(hiddenItems) {
+		this._hiddenItems = hiddenItems;
+
+		this.items.forEach((item, index) => {
+			item.hidden = index >= this._hiddenStartIndex;
+		});
+
+		this._overflowButtonText = `+${hiddenItems > 99 ? 99 : hiddenItems}`;
+	}
+
+	static get dependencies() {
```

Comment:
I would recommend moving this to the other static getters (render, template, styles, etc...)

Additionally, the component has a dependency to `ui5-button` (used in the template) and does not have  a dependency to `ui5-avatar` (at least I didn't find any reference)